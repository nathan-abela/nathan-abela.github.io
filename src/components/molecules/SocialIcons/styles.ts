import styled from 'styled-components';

const StyledSocialIcons = styled.ul`
	display: flex;

	li {
		width: 24px;
		margin: 0 8px;
		display: flex;
		justify-content: center;
	}

	a {
		color: ${({ theme }) => theme.text};
		text-decoration: none;
		transition: 0.2s;

		&:hover {
			color: ${({ theme }) => theme.headline};
		}
	}

	@media screen and (min-width: 768px) {
		li {
			margin-left: 16px;
		}
	}
`;

export { StyledSocialIcons };
