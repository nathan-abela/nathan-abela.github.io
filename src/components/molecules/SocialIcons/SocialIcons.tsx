import { FC } from 'react';
import { StyledSocialIcons } from './styles';

import { Icon } from 'Atoms/Icon';

const SocialIcons: FC = () => {
	return (
		<StyledSocialIcons>
			<li>
				<a
					href="https://github.com/nathan-abela"
					target="_blank"
					rel="noopener noreferrer"
					title="Github"
				>
					<Icon icon="GITHUB" />
				</a>
			</li>
			<li>
				<a
					href="https://www.linkedin.com/in/nathan-abela"
					target="_blank"
					rel="noopener noreferrer"
					title="LinkedIn"
				>
					<Icon icon="LINKEDIN" />
				</a>
			</li>
			<li>
				<a
					href="https://www.youtube.com/@nathanabela"
					target="_blank"
					rel="noopener noreferrer"
					title="YouTube"
				>
					<Icon icon="YOUTUBE" />
				</a>
			</li>
			<li>
				<a
					href="mailto:nathanabela7@gmail.com"
					target="_blank"
					rel="noopener noreferrer"
					title="Email"
				>
					<Icon icon="EMAIL" />
				</a>
			</li>
		</StyledSocialIcons>
	);
};

export { SocialIcons };
