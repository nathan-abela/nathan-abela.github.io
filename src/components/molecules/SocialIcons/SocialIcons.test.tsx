import { render } from '@testing-library/react';

import { SocialIcons } from './SocialIcons';

describe('Social Icons', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<SocialIcons />
		);
		expect(container).toMatchSnapshot();
	});
});
