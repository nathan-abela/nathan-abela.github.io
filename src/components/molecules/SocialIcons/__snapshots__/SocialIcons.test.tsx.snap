// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Icons renders correctly 1`] = `
<div>
  <ul
    class="styles__StyledSocialIcons-sc-1vfff3-0 fBtYTT"
  >
    <li>
      <a
        href="https://github.com/nathan-abela"
        rel="noopener noreferrer"
        target="_blank"
        title="Github"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/nathan-abela"
        rel="noopener noreferrer"
        target="_blank"
        title="LinkedIn"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="https://www.youtube.com/@nathanabela"
        rel="noopener noreferrer"
        target="_blank"
        title="YouTube"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="mailto:nathanabela7@gmail.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Email"
      >
        <test-file-stub />
      </a>
    </li>
  </ul>
</div>
`;
